Fix archive file name when downloading public share

PVince81 opened the pull request:
When download a public link share folder using the button on the top
right, it doesn't provide a list of files.

This fix makes sure to trigger the correct logic when no file list was
given.

Fixes https://github.com/owncloud/core/issues/22836

Please review @oparoz @MorrisJobke @schiesbn @rullzer

Commit:
Fix archive file name when downloading public share

When download a public link share folder using the button on the top
right, it doesn't provide a list of files.

This fix makes sure to trigger the correct logic when no file list was
given.

nickvergessen commented:
Counter proposal:
don't set files when there are none:
https://github.com/owncloud/core/blob/f79fafcf0625afdf16aad14b5584b8bad95ae1c8/apps/files_sharing/js/public.js#L162-L162

Then this matches the smashbox test we have in place.
You set `files: undefined` here, so it sets the thing instead of sending null (not sending at all)

PVince81 commented:
Why not both ? :wink: 

I'll have a look at the JS code.

Commit:
Remove "files" arg from download URL in public link page

When no files were specified for download, it means folder download.
In this case, no need to pass an empty "files" argument.

PVince81 commented:
@nickvergessen https://github.com/owncloud/core/commit/dbea268332aef3c9f3053f9fe55fbaa9ea43b7d8

nickvergessen commented:
:+1: should backport to 9.0.1?

PVince81 commented:
Not critical, but could be a user annoyance.

@karlitschek backport to 9.0.1 and 8.2.4 ?

rullzer commented:
:+1:

jancborchardt commented:
(I’d say backport is important since it’s a regression.)

nickvergessen commented:
its only the name of the zip, so nothing really important. most people will unzip anyway and the name of the folder is good again. So annoying yes, important no.

karlitschek commented:
please backport :+1:

jancborchardt commented:
@nickvergessen _very_ annoying when downloading multiple folders.

PVince81 commented:
stable9: https://github.com/owncloud/core/pull/22922
stable8.2: https://github.com/owncloud/core/pull/22923
